patches_for_WebRTC_org/m80/0007.3-Fixing-NV12-to-I420-conversion-with-strides-and-gap.patch

```diff
@@ -0,0 +1,76 @@
+From 144df9fb8e4dbc56130b276291ee0d8513313c81 Mon Sep 17 00:00:00 2001
+From: Augusto Righetto <aurighet@microsoft.com>
+Date: Tue, 9 Jun 2020 08:59:05 -0700
+Subject: [PATCH] Fixing NV12 to I420 conversion with strides and gap
+
+---
+ include/libyuv/convert.h  |  3 +++
+ source/convert_to_i420.cc | 19 ++++++++++---------
+ 2 files changed, 13 insertions(+), 9 deletions(-)
+
+diff --git a/include/libyuv/convert.h b/include/libyuv/convert.h
+index f571142f..b4534f14 100644
+--- a/include/libyuv/convert.h
++++ b/include/libyuv/convert.h
+@@ -481,6 +481,9 @@ int MJPGSize(const uint8_t* sample,
+ LIBYUV_API
+ int ConvertToI420(const uint8_t* sample,
+                   size_t sample_size,
++                  int src_stride_y,
++                  const uint8_t* src_uv,
++                  int src_stride_uv,
```

Comment:
See Jerome's comment, we should use `NV12ToI420()` and `NV12ToI420()` if we know these are the only formats that we are going to get. If we need to support other formats, we can use the explicit conversion functions for the ones that may have separate Y and UV planes (dunno if there are others apart from NV12 and NV21) and the default `ConvertToI420` for all the others.

This specific change has issues that make it viable just as a short-term workaround (breaks the API of libyuv; adds parameters that are useful only to specific formats to a generic function; maybe implementation is incomplete as Jerome pointed out). Also note that `ConvertToI420` already takes both `src_width` and `crop_x` that can be used as stride and width in general.